### ISSUE TITLE
Enable running downstream op even if upstream op fails

### ISSUE THREAD

**Issue by sryza** [dagster-io/dagster#15761]
### What we've heard
- Wenlan Wei - https://dagster.slack.com/archives/C02LJ7G0LAZ/p1691519305546259
  - Hi team, Is there anyway that I could customize the trigger rule in Dagster. Like If I have ops and a downstream ops, is there anyway that I could still trigger the ops even if the upstream has a failure.
  - In this case, I am using all non-input task, so there is no dependencies on the input data.
  - ![image](https://github.com/dagster-io/dagster/assets/654855/1c000b65-9344-445c-93fb-390ec62f4a91)

**Comment by NicolasGuary** [comment]
Hello, any news on this topic @sryza ? 🙏 

As a user, I think that what would make sense would be to have the ability mark a specific Asset as non-blocking. So, in your example, you could mark the **Asset** associated to **Op4** as non-blocking, and **Downstream** would still materialize even if it fails.
But, the **Asset** associated to **Op2** could not be flagged, and so **Downstream** wouldn't materialize if it fails (+ job status reported would be failure).

**Comment by girguy** [comment]
I totally agree with @NicolasGuary. I am highly expected this feature to be available at some point

**Comment by MfMark** [comment]
We also need this feature.

**Comment by maxwellwood** [comment]
we also need this feature!
